gulpfile.js/build.js

```diff
@@ -0,0 +1,411 @@
+    (new SpecImporter()).import(),
+    // TODO: Fails on Travis with HttpError: Requires authentication
+    // roadmapImporter.importRoadmap(),
+  ]);
+}
+
+
+/**
+ * Builds playground and boilerplate generator, imports all remote documents,
+ * builds samples, lints Grow pod and JavaScript.
+ *
+ * @return {undefined}
+ */
+function setupBuild(done) {
+  gulp.series(
+      test.lintNode,
+      // Build playground and boilerplate that early in the flow as they are
+      // fairly quick to build and would be annoying to eventually fail downstream
+      gulp.parallel(buildPlayground, buildBoilerplate, buildSamples, importAll),
+      // TODO: Fix working but malformatted references before reenabling
+      // test.lintGrow,
+      async function storeArtifcats() {
+        if (!travis.onTravis() ||
+            (travis.repo.branch && !BUILDING_BRANCHES.includes(travis.repo.branch))) {
+          return;
+        }
+
+        // If on Travis store everything built so far for later stages to pick up
+        // Local path to the archive containing artifacts of the first stage
+        const SETUP_ARCHIVE = 'build/setup.tar.gz';
+        // All paths that contain altered files at build setup time
+        const SETUP_STORED_PATHS = [
+          './pages/content/',
+          './dist/',
+          './boilerplate/dist/',
+          './playground/dist/',
+          './.cache/',
+          './examples/static/samples/samples.json',
+        ];
+
+        await sh('mkdir -p build');
+        await sh(`tar cfj ${SETUP_ARCHIVE} ${SETUP_STORED_PATHS.join(' ')}`);
+        await sh(`gsutil cp ${SETUP_ARCHIVE} ` +
+          `${TRAVIS_GCS_PATH}${travis.build.number}/setup.tar.gz`);
+      })(done);
+}
+
+/**
+ * Fetches remote artifacts that have been built in earlier stages
+ *
+ * @return {Promise}
+ */
+async function fetchArtifacts() {
+  await sh('mkdir -p build');
+  if (travis.onTravis() || config.options['travis-build']) {
+    const buildNumber = config.options['travis-build'] || travis.build.number;
+    await sh(`gsutil cp -r ${TRAVIS_GCS_PATH}${buildNumber} ${project.paths.BUILD}`);
+    await sh('find build -type f -exec tar xf {} \;');
+  }
+}
+
+/**
+ * Starts Grow to build the pages
+ *
+ * @return {Promise}
+ */
+async function buildPages() {
+  config.configureGrow();
+  await sh('grow deploy --noconfirm --threaded', {
+    workingDir: project.paths.GROW_POD,
+  });
+
+  await transformPages();
+
+  // ... and again if on Travis store all built files for a later stage to pick up
+  if (travis.onTravis()) {
+    const archive = `build/pages-${travis.build.job}.tar.gz`;
+    await sh(`tar cfj ${archive} ./dist/pages`);
+    await sh(`gsutil cp ${archive} ` +
+      `${TRAVIS_GCS_PATH}${travis.build.number}/pages-${travis.build.job}.tar.gz`);
+  }
+}
+
+/**
+ * Transforms already built pages and does so while spawning multiple child
+ * processes to speed up processing
+ *
+ * @return {Promise}
+ */
+async function transformPages() {
```

Comment:
This task got a bit ugly as I tried to introduce sharding. I thought of it having a bigger effect, but it only saves ~10 seconds per locale. But the Travis VMs also only run 2 cores.

Reply:
That's really surprising. What's the bottleneck?